Sources/App/Controllers/API/TodoController.swift

```diff
@@ -0,0 +1,33 @@
+import Vapor
+import FluentSQLite
+
+/// Controls basic CRUD operations on `Todo`s.
+final class TodoController {
+    /// Returns a list of all `Todo`s.
+    func index(_ req: Request) throws -> Future<[Todo]> {
+        let repository = try req.make(TodoRepository.self)
+
+        var criteria: [FilterOperator<Todo.Database, Todo>] = []
+
+        if let title = req.query[String.self, at: "title"] {
+            criteria.append(.make(\Todo.title, .equal, [title]))
+        }
+
+        return repository.findBy(criteria: criteria, orderBy: nil, on: req)
```

Comment:
I'm not sure I'm a fan of the making this overload an array of criteria. I understand the intent but I wonder if there's a better way to do it. Maybe something with variadics? Not a deal breaker for me but it feels ugly to have to construct an array of generic `FilterOperators` to make a database call

Reply:
mmmm the problem of variadic i think it is the KeyPath Value...

For now i think that is better than having functions like `filterByTitle`, `FilterByBody`, `FilterByTitleAndBody` or their equivalent using `filter(title:, body:)` with optional values

With small models can work, but with models with many properties maybe is not a good idea

Reply:
Yeah I’m fine with that! Will merge now